File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/account/infrastructure/persistence/SpendingForecastJpaRepository.java
package com.example.accountservice.account.infrastructure.persistence;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SpendingForecastJpaRepository
        extends JpaRepository<SpendingForecastJpaEntity, Long> {

    boolean existsByAccountIdAndForecastMonth(String accountId, String forecastMonth);

    Optional<SpendingForecastJpaEntity> findByAccountIdAndForecastMonth(
            String accountId, String forecastMonth);
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/account/infrastructure/persistence/SpendingForecastJpaRepository.kt
package com.example.accountservice.account.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository

interface SpendingForecastJpaRepository : JpaRepository<SpendingForecastJpaEntity, Long> {
    fun existsByAccountIdAndForecastMonth(
        accountId: String,
        forecastMonth: String,
    ): Boolean

    fun findByAccountIdAndForecastMonth(
        accountId: String,
        forecastMonth: String,
    ): SpendingForecastJpaEntity?

    // Used only by TaskQueueE2ETest to assert re-enqueueing the same month never produces a second row.
    fun countByAccountIdAndForecastMonth(
        accountId: String,
        forecastMonth: String,
    ): Long
}
